src/jackdaw/admin.clj

```diff
@@ -6,11 +6,54 @@
   Like the underlying `AdminClient` API, this namespace is subject to
   change and should be considered of alpha stability."
   {:license "BSD 3-Clause License <https://github.com/FundingCircle/jackdaw/blob/master/LICENSE>"}
-  (:require [jackdaw.data :as jd])
-  (:import [org.apache.kafka.clients.admin
-            AdminClient DescribeTopicsOptions]))
-
-(set! *warn-on-reflection* true)
+  (:require
+   [jackdaw.data :as jd]
+   [manifold.deferred :as d])
+  (:import [org.apache.kafka.clients.admin AdminClient
+            DescribeTopicsOptions DescribeClusterOptions DescribeConfigsOptions]))
+
+(defprotocol Client
+  (alter-topics* [this topics])
+  (create-topics* [this topics])
+  (delete-topics* [this topics])
+  (describe-topics* [this topics])
+  (describe-configs* [this configs])
+  (describe-cluster* [this])
+  (list-topics* [this]))
+
+(def client-impl
+  {:alter-topics* (fn [this topics]
+                   (d/future
+                     (->> (.alterConfigs this topics)
+                          .all deref)))
```

Comment:
On style, is `(deref (.all (.alterConfigs this topics)))` or `@(.all (.alterConfigs this topics))` actually clearer here? I _personally_ find it easier than the threading macro version (but I have never got on with the threading macro anyway ...).

Reply:
+1